src/hal/src/device.rs

```diff
@@ -758,4 +758,42 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
     ///
     /// Host access to all queues needs to be **externally** sycnhronized!
     fn wait_idle(&self) -> Result<(), OutOfMemory>;
+
+    /// Associate a name with an image, for easier debugging in external tools or with validation
+    /// layers that can print a friendly name when referring to objects in error messages
+    unsafe fn set_image_name(&self, image: &B::Image, name: &str) -> Result<(), OutOfMemory>;
+    /// Associate a name with a buffer, for easier debugging in external tools or with validation
+    /// layers that can print a friendly name when referring to objects in error messages
+    unsafe fn set_buffer_name(&self, buffer: &B::Buffer, name: &str) -> Result<(), OutOfMemory>;
```

Comment:
it looks like Vulkan [requires external synchronization](https://vulkan.lunarg.com/doc/view/1.0.30.0/linux/vkspec.chunked/ch32s01.html#vkDebugMarkerSetObjectNameEXT) for all the objects that we need to set names for:
> Host access to pNameInfo.object must be externally synchronized 

That means we should have `&mut B::Buffer` (and same for other methods). On the good side, the implementations don't need to have `Mutex` in them.